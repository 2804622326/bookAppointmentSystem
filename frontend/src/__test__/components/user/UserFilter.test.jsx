import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserFilter from '../../../components/user/UserFilter.jsx';

test('selects value and clears filter', () => {
  const onSelect = jest.fn();
  const onClear = jest.fn();
  const { asFragment } = render(
    <UserFilter label='Role' values={['Admin', 'User']} selectedValue='' onSelectedValue={onSelect} onClearFilters={onClear} />
  );
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Admin' } });
  expect(onSelect).toHaveBeenCalledWith('Admin');
  fireEvent.click(screen.getByRole('button'));
  expect(onClear).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
