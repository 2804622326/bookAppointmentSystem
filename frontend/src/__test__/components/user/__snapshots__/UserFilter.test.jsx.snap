// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selects value and clears filter 1`] = `
<DocumentFragment>
  <div
    class="mb-2 input-group"
  >
    <span
      class="input-group-text"
    >
      Filter by Role
    </span>
    <select
      class="form-control form-select"
    >
      <option
        value=""
      >
        ...Select role...
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="User"
      >
        User
      </option>
    </select>
    <button
      class="btn btn-secondary"
      type="button"
    >
      Clear Filter
    </button>
  </div>
</DocumentFragment>
`;
